misc/tests/conftest.py

```diff
@@ -0,0 +1,8 @@
+def pytest_addoption(parser):
+    parser.addoption("--suppress", action="append", default=[],
+                     help="Suppression files used in test-memory-leaks.py. These files must be located in the folder valgrind.")
+
+def pytest_generate_tests(metafunc):
+    if "suppression_files" in metafunc.fixturenames:
+        tags = list(metafunc.config.option.suppress)
+        metafunc.parametrize("suppression_files", [tags], scope='session')
```

Comment:
Wouldn't `[tags]` wrap the argument in an additional list, so we end up with a list of lists? If the parameter `metafunc.config.option.suppress` is a single string and not already some collection, then the issue is the `list(...)` in line 7, instead.